ops/pipeline/build-cuda13.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+## Build XGBoost with CUDA 13
+
+set -euo pipefail
+
+if [[ -z "${GITHUB_SHA:-}" ]]
+then
+  echo "Make sure to set environment variable GITHUB_SHA"
+  exit 1
+fi
+
+IMAGE_REPO="xgb-ci.gpu_build_cuda13_rockylinux8"
+export USE_RMM=0
+export USE_FEDERATED=0
+
+source ops/pipeline/classify-git-branch.sh
+source ops/pipeline/get-docker-registry-details.sh
+source ops/pipeline/get-image-tag.sh
+
+WHEEL_TAG=manylinux_2_28_x86_64
+BUILD_IMAGE_URI="${DOCKER_REGISTRY_URL}/${IMAGE_REPO}:${IMAGE_TAG}"
+MANYLINUX_IMAGE_URI="${DOCKER_REGISTRY_URL}/xgb-ci.${WHEEL_TAG}:${IMAGE_TAG}"
+
+echo "--- Build with CUDA"
+
+if [[ ($is_pull_request == 1) || ($is_release_branch == 0) ]]
+then
+  export BUILD_ONLY_SM75=1
+else
+  export BUILD_ONLY_SM75=0
+fi
+
+set -x
+
+# Remove nvidia-nccl-cu12 from the list of Python deps
+# nvidia-nccl-cu13 is not yet available on PyPI
```

Comment:
@hcho3 even when `nvidia-nccl-cu13` has wheels up on PyPI, what will be the plan for `xgboost` wheels?

Back in #10807 we'd talked about ways to support multiple major CUDA versions at the same time but I don't believe anything was implemented for that, right?

Reply:
We will need to create a new namespace `xgboost-cu13`. This can be done pretty quickly, since the wheel size is under 100 MB now.

Reply:
Updated proposal: https://github.com/dmlc/xgboost/issues/10807#issuecomment-3255218348